src/Samples/SubModelSeq.List/Program.fs

```diff
@@ -0,0 +1,121 @@
+module Elmish.WPF.Samples.SubModelSeq.List.Program
+
+open Elmish
+open Elmish.WPF
+
+
+type InOutMsg<'a, 'b> =
+  | InMsg of 'a
+  | OutMsg of 'b
+
+
+module List =
+
+  let cons head tail = head :: tail
+
+  let mapAtIndex idx f =
+    List.indexed >> List.map (fun (i, a) -> if i = idx then f a else a)
+
+  let removeAtIndex idx ma =
+    [ ma |> List.take idx
+      ma |> List.skip (idx + 1)
+    ] |> List.concat
+
+
+[<AutoOpen>]
+module Counter =
+
+  type Counter =
+    { Count: int
+      StepSize: int }
+
+  type CounterMsg =
+    | Increment
+    | Decrement
+    | SetStepSize of int
+    | Reset
+
+  module Counter =
+
+    let init =
+      { Count = 0
+        StepSize = 1 }
+    
+    let canReset = (<>) init
+    
+    let update msg m =
+      match msg with
+      | Increment -> { m with Count = m.Count + m.StepSize }
+      | Decrement -> { m with Count = m.Count - m.StepSize }
+      | SetStepSize x -> { m with StepSize = x }
+      | Reset -> init
+
+    let bindings () = [
+      "CounterValue" |> Binding.oneWay(fun (_, _, c) -> c.Count)
+      "Increment" |> Binding.cmd(Increment |> InMsg)
+      "Decrement" |> Binding.cmd(Decrement |> InMsg)
+      "StepSize" |> Binding.twoWay(
+        (fun (_, _, c) -> float c.StepSize),
+        (fun v _ -> v |> int |> SetStepSize |> InMsg))
+      "Reset" |> Binding.cmdIf(
+        Reset |> InMsg,
+        (fun (_, _, c) -> canReset c))
+    ]
+
+
+module App =
+
+  type Model = Counter list
```

Comment:
I suggest having the main `Model` be a record with a single field `Counters`. When I learned Elmish, I was always annoyed at the countless (har-har) counter tutorials where the model was just an `int`; it made the rest of the code artificially simple and a bit harder to understand.

Reply:
Sure.  Done.